### PR TITLE
feat: labels on editor, proposal and proposal list

### PR DESCRIPTION
**Pull request by ChaituVR**
Towards https://github.com/snapshot-labs/workflow/issues/143

### Summary
- Labels selector on editor page 
Untitled 7

- Labels on proposal and proposal list
Untitled 4
Untitled 5

### How to test

1. Add some labels in your space using https://github.com/snapshot-labs/sx-monorepo/pull/829
2. You should see labels selector on editor similar to here https://www.figma.com/design/hmHTau1Y1KYzrBBKLQr5QV/%23design?node-id=7384-1834&node-type=frame&m=dev
3. Should be able to save labels on proposals
4. You should be able to view labels on proposal and proposal list

### CODE REVIEW

**Comment by bonustrack** [comment]
- Dropdown should appear on top of timeline and not expand the sidebar content.
- There should be some space between labels if they appear on 2 different line
image
- We should truncate label description in the editor selector
- Use `text-sm` for description

File: apps/ui/src/views/Proposal.vue [hunk 1]
```diff
@@ -320,6 +320,9 @@ watchEffect(() => {
               :decimals="votingPowerDecimals"
             />
           </div>
+          <div v-if="space.labels?.length && proposal.labels?.length">
```
Comment (wa0x6e) [inline apps/ui/src/views/Proposal.vue:323]: The wrapping div may be redundant, and we could apply the `if` directly on `ProposalLabels`

File: apps/ui/src/views/Space/Editor.vue [hunk 2]
```diff
@@ -495,6 +507,14 @@ watchEffect(() => {
           >.
         </template>
       </EditorChoices>
+      <div v-if="space.labels?.length">
```
Comment (wa0x6e) [inline apps/ui/src/views/Space/Editor.vue:510]: Same here, wrapping div is redundant

File: apps/ui/src/components/ProposalLabels.vue [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      <button
+        v-if="showEdit"
+        type="button"
+        class="h-0"
+        @click="labelSelectorOpen = true"
+      >
+        <IH-pencil />
+      </button>
+    </div>
+    <div
```
Comment (wa0x6e) [inline apps/ui/src/components/ProposalLabels.vue:70]: Is it possible to use `Combobox` here ?

**Review by Sekhmet — CHANGES_REQUESTED** [review]
Needs changeset.

File: packages/sx.js/src/clients/offchain/types.ts [hunk 4]
```diff
@@ -169,6 +169,7 @@ export type Propose = {
   body: string;
   discussion: string;
   choices: string[];
+  labels: string[];
```
Comment (Sekhmet) [inline packages/sx.js/src/clients/offchain/types.ts:172]: Don't we need to update those as well?
https://github.com/snapshot-labs/sx-monorepo/blob/7076db1633d69b7018f49f6a10e62ced9fb74a7a/packages/sx.js/src/clients/offchain/ethereum-sig/types.ts#L62-L78

https://github.com/snapshot-labs/sx-monorepo/blob/36bdb33f375f127a043cca574fe44fd1edf9c088/packages/sx.js/src/clients/offchain/types.ts#L52-L66

File: apps/ui/src/components/ProposalLabels.vue [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+            <IH-check class="text-skin-success" />
+          </div>
+        </div>
+        <div
+          v-if="filteredLabels.length === 0"
+          class="px-3 py-[11.5px] bg-skin-border"
+        >
+          No results
+        </div>
+      </div>
```
Comment (Sekhmet) [inline apps/ui/src/components/ProposalLabels.vue:113]: Those are not really accessible, you can't select options with keyboard.

Have you considered using Headless UI's Combobox for this? We would also make it float above content as mentioned in the call.

File: apps/ui/src/components/ProposalLabels.vue (hunk 3 above)
Comment (ChaituVR) [reply apps/ui/src/components/ProposalLabels.vue:70]: Need some refactoring/add many conditions to handle the new design. don't think it worth, wyt @Sekhmet 

Combobox:
image
LabelSelctor
image

Comment (ChaituVR) [reply apps/ui/src/components/ProposalLabels.vue:70]: Looks like for accessbility, i will look into it

Comment (Sekhmet) [reply apps/ui/src/components/ProposalLabels.vue:70]: I wouldn't use our Combobox directly, but rather build it using raw Combobox from headless UI:
https://headlessui.com/v1/vue/combobox

**Review by Sekhmet — APPROVED** [review]
tACK

**Pull request merged**